pkg/client/transaction.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2021 CodeNotary, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"github.com/codenotary/immudb/pkg/api/schema"
+	"github.com/codenotary/immudb/pkg/client/errors"
+	"github.com/golang/protobuf/ptypes/empty"
+)
+
+type TxOptions struct {
+	TxMode schema.TxMode
+}
+
+type Tx interface {
+	Commit(ctx context.Context) (*schema.CommittedSQLTx, error)
+	Rollback(ctx context.Context) error
+
+	TxSQLExec(ctx context.Context, sql string, params map[string]interface{}) error
+	TxSQLQuery(ctx context.Context, sql string, params map[string]interface{}, renewSnapshot bool) (*schema.SQLQueryResult, error)
```

Comment:
`renewSnapshot` or `ReuseSnapshot ` should be removed on the endpoints using a specific Tx